.github/workflows/ci.yml

```diff
@@ -300,46 +301,100 @@ jobs:
       - name: Build
         run: cmake.exe --build C:\build --config Release --verbose
 
-      - name: Test with the first try
-        id: first-try
-        run: |
-          cd C:\build 
-          ctest.exe --output-on-failure
-        continue-on-error: true
-
-      - name: Test with the second try for failed cases
-        id: second-try
-        if: ${{ steps.first-try.outcome == 'failure' }} 
-        run: |
-          cd C:\build 
-          ctest.exe --rerun-failed --output-on-failure
-        continue-on-error: true
-
-      - name: Test with the third try for failed cases
-        id: third-try
-        if: ${{ steps.second-try.outcome == 'failure' }} 
-        run: |
-          cd C:\build 
-          ctest.exe --rerun-failed --output-on-failure
-        continue-on-error: true
-
-      - name: Test with the fourth try for failed cases
-        id: fourth-try
-        if: ${{ steps.third-try.outcome == 'failure' }} 
-        run: |
-          cd C:\build 
-          ctest.exe --rerun-failed --output-on-failure
-        continue-on-error: true
+      - name: Zip build folder
+        run: 7z.exe a C:\build.7z "C:\build"
 
-      - name: Test with the last try for failed cases
-        if: ${{ steps.fourth-try.outcome == 'failure' }} 
-        run: |
-          cd C:\build 
-          ctest.exe --rerun-failed --output-on-failure
+      - uses: actions/upload-artifact@v3
+        with:
+          name: Windows-build
+          path: C:\build.7z
+          if-no-files-found: error
+          retention-days: 1
 
+  Windows-test:
+    needs: Windows-build
+    runs-on: windows-latest
+    env:
+      VCPKG_DEFAULT_TRIPLET: x64-windows
+        
+    steps:
+     - uses: actions/checkout@v3 # Checks-out the repository under ${{github.workspace}}
+              
+     - name: Update ccache and ninja # For correct caching with ccache on Windows
+       shell: bash
+       run: |
+         choco install ccache
+         choco install ninja
+         choco install 7zip
+
+     - uses: hendrikmuhs/ccache-action@v1.2
+       with:
+         key: ${{ github.job }}
+        
+     - uses: friendlyanon/setup-vcpkg@v1 # Setup vcpkg into ${{github.workspace}}
+       with: 
+         committish: ${{ env.VCPKG_VERSION }}
+         cache-version: ${{env.VCPKG_VERSION}}
+        
+     - name: Install dependencies
+       run: |
+         ${{github.workspace}}\vcpkg\vcpkg.exe install --clean-after-build `
+           eigen3 `
+           tbb `
+           boost-program-options `
+           boost-geometry `
+           simbody `
+           gtest `
+           xsimd `
+           pybind11 
+        
+     - uses: actions/download-artifact@v3
+       with:
+         name: Windows-build
```

Comment:
You must precise where to download it, by default it is not downloaded in `C:`, but in `${{github.workspace}}`

Reply:
I use it by follow the old Windows CI.

Reply:
Or the extraction must specify the input path and target location. Probably something along those lines:
```yaml
     - name: Unzip build folder
       run: 7z.exe e ${{github.workspace}}\build.7z C:
```